Hippo.Web/ClientApp/src/components/Order/Details.tsx

```diff
@@ -0,0 +1,685 @@
+                  const model: PaymentModel = {
+                    ...editPaymentModel,
+                    amount: parseFloat(value),
+                    entryAmount: value,
+                  };
+                  setEditPaymentModel(model);
+                  setReturn(model);
+                }
+              }}
+            />
+          </div>
+        </>
+      ),
+      canConfirm: !notification.pending && editPaymentModel.amount > 0,
+    },
+    [editPaymentModel, notification.pending],
+  );
+
+  const makePayment = useCallback(async () => {
+    const [confirmed, editPaymentModel] = await makePaymentConfirmation();
+
+    if (!confirmed) {
+      return;
+    }
+    //console.log(editPaymentModel);
+
+    const req = authenticatedFetch(
+      `/api/${cluster}/order/makepayment/${orderId}?amount=${editPaymentModel.amount}`,
+      {
+        method: "POST",
+      },
+    );
+    setNotification(req, "Making Payment", "Payment Made", async (r) => {
+      if (r.status === 400) {
+        const errors = await parseBadRequest(response);
+        return errors;
+      } else {
+        return "An error happened, please try again.";
+      }
+    });
+
+    const response = await req;
+
+    if (response.ok) {
+      const data = await response.json();
+      console.log(data);
+      setOrder(data);
+      setEditPaymentModel({
+        id: 0,
+        amount: 0,
+        status: "",
+        createdOn: "",
+        entryAmount: "",
+      });
+      calculateBalanceRemaining(data);
+    }
+  }, [cluster, orderId, makePaymentConfirmation, setNotification]);
+
+  const [approveOrderConfirmation] =
+    useConfirmationDialog<UpdateOrderStatusModel>(
+      {
+        title: "Update Order Status",
+        message: (setReturn) => (
+          <>
+            <div>Current Status: {updateStatusModel.currentStatus}</div>
+            <div>Set Status to: {updateStatusModel.newStatus}</div>
+            <hr />
+            {updateStatusModel.newStatus === "Submitted" && (
+              <div style={{ backgroundColor: "#90ee90" }}>
+                This will submit the order to the cluster admins for processing.
+              </div>
+            )}
+            {updateStatusModel.newStatus === "Processing" && (
+              <div style={{ backgroundColor: "#90ee90" }}>
+                This will indicate that an admin will start working on the
+                order.
+              </div>
+            )}
+            {updateStatusModel.newStatus === "Active" && (
+              <div style={{ backgroundColor: "#90ee90" }}>
+                This will move the order to active and allow manual billing as
+                well as scheduled billing.
+              </div>
+            )}
+          </>
+        ),
+        canConfirm: !notification.pending,
+      },
+      [order, notification.pending, updateStatusModel],
+    );
+
+  const updateStatus = useCallback(async () => {
```

Comment:
why useCallback here?  why here and not reject or cancel methods:?

Reply:
Can't remember for sure, but I think I was having trouble getting dynamic values for the status to work, and I just wanted one place where the status would get changed.